logger.h

```diff
@@ -0,0 +1,19 @@
+#include <mutex>
+#include <iostream>
+
+namespace Logger {
+	static std::mutex print_lock;
+
+	inline void log_info(const std::string &msg, const bool std_out = true) {
```

Comment:
All log methods have the same signature and contents, so I'd rather have a single log function that takes the level as parameter. `std_out` is unused, so I'd remove it. It's a header only inline function so we can add a default parameter later anyway

Reply:
I agree with removing currently uneeded stdout flag param. However I think it's better for usability to keep the levels as different functions. This "function per log level" is pretty standard in logging libraries.